### ISSUE TITLE
Not working on Index

### ISSUE THREAD

**Issue by tnup** [epartment/nova-dependency-container#92]
Is this intended behaviour, or is this a bug?

```
 NovaDependencyContainer::make([
                Text::make("status")->showOnIndex(),
            ])->dependsOnNotEmpty('validformat', true),
```

will work on details and edit, but not on Index.

**Comment by wize-wiz** [comment]
@tnup This was indeed intended by the author of this package. I'm working on a newer version that includes all default field behaviours. Maybe I can port some of the new stuff into the current 1.2 branch, I'll let you know.

**Comment by tnup** [comment]
dankjewel ;-) thanks!

**Comment by msalmanshahid** [comment]
I am facing this too. Any update please?

**Comment by vFire** [comment]
Me too...

**Comment by abensoukehal** [comment]
@wize-wiz any update please ?

**Comment by vFire** [comment]
Finally, I know why it can't support showOnIndex, if it did, there will be different number or the table column and in the same col, there will be different field and value.
So ,we can easily add another computed fields and show it on index only.

**Comment by goromachine** [comment]
could you please show us an example of what you are saying, @vFire ? I would like to show on the index the field I'm filling inside a NovaDependencyContainer.

Thanks

**Comment by abensoukehal** [comment]
(Workaround while waiting for the PR) You can do that by duplicate the field you want to show and smake it show only on index (`->onlyOnIndex`), 
```
Select::make(__('Offset'), 'offset')->options([
          'before' => 'Before',
          'immediately' => 'Immediately',
          'after' => 'After'
])->displayUsingLabels()
          ->rules('required'),
```

```
NovaDependencyContainer::make([
        Select::make(__('Offset Interval'), 'offset_interval')->options([
            'hour' => 'Hour',
            'day' => 'Day',
            'week' => 'Week',
            'month' => 'Month',
        ])->displayUsingLabels()
            ->rules('required'),
        Number::make(__('Offset period'), 'offset_period')
            ->rules('required'),
])->dependsOn('offset', 'before')->dependsOn('offset', 'after'),
```

```
Number::make(__('Offset period'), 'offset_period')
        ->rules('required')
        ->onlyOnIndex()
```

**Comment by goromachine** [comment]
@abensoukehal thank you!

Another question: In case of wanting to show the select 'offset interval' too, I should add it outside the NovaDependencyContainer with the same options right? If that is true, is there any way of not duplicating the options to show them on both sides?in my case the options are something like `'CP' => 'Coupe', 'OR' => 'OffRoad'` and many more.

Thanks!

**Comment by abensoukehal** [comment]
Yes, unfortunately @goromachine, you have to duplicate outside `NovaDependencyContainer` all the fields you want to show on the index view, and of course, make them show onlyOnIndex().

@wize-wiz any update ?
